docs/jumpstart/week-10.md

```diff
@@ -0,0 +1,221 @@
+```shell 
+heroku login
+```
+follow the instructions to enter our Heroku credentials. Once we have logged in, we now have to create a new Heroku project. 
+
+To create a project we type the command:
+```shell 
+heroku create (PROJECT_NAME) --buildpack mars/create-react-app
+```
+By doing this, Heroku is creating a new deployment server for us and giving us a custom URL for us to use. The buidlpack part is our way of asserting
+that our project is a React project. Usually, Heroku automatically detects that it is a React application but sometimes it does not. We add the buildpack just to avoid any errors
+in case. Now comes the part where we connect our application with our new Heroku created project. 
+```shell
+git remote add heroku
+```
+This will add Heroku to our git repository which is how we will push our code to Heroku. Finally, to push our code we run the command: 
+```shell
+git push heroku master
+```
+This will push our current committed code that is in our <strong> master </strong> branch on git. If we want to deploy a branch that is not 
+the master's branch we can run the command:
+```shell
+git push heroku <branch_name>:master
+```
+Now that our Frontend code is uploaded to Heroku, we can use the url Heroku gave us to open our application online. Now, anyone with this link will be able to see our application.
+To open our application we can run:
+```shell 
+heroku open
+```
+This should open a browser tab with our project's home page. 
+However, this is only the user interface portion of our application. Now we have to deploy our backend code.
+
+
+### Deploy backend
+To deploy our backend application, first we have to configure our application. First, we have to navigate to our pom.xml 
+file which gets created after every maven project. We have to configure all our app's dependencies through maven so that when we export our Java project our dependencies also get exported.
+An app's dependencies is external code that other people wrote that we use in our application. Usually, if we did not have problems during local development, then our pom.xml file should be fine and ready to be exported.
+If not, then now is the time to fix it. Potential fixes include using different versions of a dependency or getting rid of dependencies that we did not end up using.
+Next, we have to create a Procfile. This is a text-like file that that acts almost like a configuration file that Heroku uses to set up your java application. Our procfile must look like:
+
+```ignorelang
+web: java $JAVA_OPTS -cp target/classes:target/dependency/* <project main package>
+```
+
+In our case, if you download our catch-up code, our procfile looks like:
+```ignorelang
+web: java $JAVA_OPTS -jar service/target/service-1.0-SNAPSHOT-jar-with-dependencies.jar
+```
+
+What this procfile is doing is specifying that our application will use the HTTP protocol, indicated by the "web" portion and our "java" section specifies it is a Java application. 
```

Comment:
```suggestion
This Procfile specifies that our application will use the web, indicated by the "web" portion, and our "java" section specifies how to start it up. 
```

The "web" part lets it know it's not just a background process, and that it should be allowed to receive/make requests